src/components/table/examples/table-selectable-rows.tsx

```diff
@@ -0,0 +1,61 @@
+import { Component, h, State } from '@stencil/core';
+import { Column } from '../table.types';
+import { persons, Person } from './persons';
+
+/**
+ * Select rows
+ *
+ * @link persons.ts
+ */
+@Component({
+    tag: 'limel-example-table-selectable-rows',
+    styleUrl: 'table.scss',
+    shadow: true,
+})
+export class TableExampleSelectableRows {
+    @State()
+    private activeRow: Person;
```

Comment:
@anderssonjohan asked:
> When activating a row and then clicking its checkbox, should that clear the activeRow immediately or when a second item is checked? If not clearing this immediately it will always open up the limeobject preview dialog when making the first selection. See screenrec for example of selecting the activated item and deactivating it while it remains visually active. 

We're not sure what activeRow exactly refers to. If by "should that clear the activeRow immediately" you mean removing the green background highlight? The answer would be YES. 

But if you mean "closing the related view", we believe currently believe that the related view should remain open. The green background highlight should only show which item is open in related view. This interaction should remain independent, if possible. You can get a feel for this interaction if you play around in Lime Go.

> One solution could be to use one visual style for activation and another for selection?

We think a nice and visually noticeable checkbox (not Tabulator's default checkboxes) is enough as a distinct and different visual style.

Reply:
> If by "should that clear the activeRow immediately" you mean removing the green background highlight? The answer would be YES.
>
> But if you mean "closing the related view", we believe currently believe that the related view should remain open. The green background highlight should only show which item is open in related view.

Sounds to me like you're contradicting yourself here, so either I misunderstood what I read, or you didn't write what you meant 😄

1. You say that the green background highlight should be removed
2. You say that the related view should not be closed
3. You say that a line should be green when (and only when), that item is open in the related view

---

Personally, I think that clicking a checkbox shouldn't change the status of the related-view at all. Whether checking or unchecking a checkbox, that interaction should neither open nor close the related view, and it also shouldn't change what's shown in the related view, if it's already open and showing something else.

Reply:
Roger roger!

1. Then I will remove the part emitting null for activeRow in the selection handler.
2. It is up to the consumer of limel-table to do what they want by using `onActivate` and `onSelect`
3. The data represented by `activeRow` may or may not be in `selection`
4. I'll update the `formatRow` to use the highlight css class `active` for the _selected rows_ only when there's a selection. Thus, it will only highlight `activeRow` when there's no selection; `activeRow` is not always represented visually by having the `active` css class applied to it.

Reply:
> 1. Then I will remove the part emitting null for activeRow in the selection handler.
> 2. It is up to the consumer of limel-table to do what they want by using `onActivate` and `onSelect`
> 3. The data represented by `activeRow` may or may not be in `selection`
> 4. I'll update the `formatRow` to use the highlight css class `active` for the _selected rows_ only when there's a selection. Thus, it will only highlight `activeRow` when there's no selection; `activeRow` is not always represented visually by having the `active` css class applied to it.

I agree with all of those, except point 4. `activeRow` should always be highlighted if there is one, the rows in `selection` should never affect the highlighting. The `selection` status is communicated by the checkboxes.

Reply:
Either the `active` css class shouldn't be set on selected rows, or, if that's set automatically by Tabulator, we should add a separate css class for applying the highlighting for the active row.

Reply:
Aaah… crap… the `activeRow` uses Tabulator's row selection feature… now I understand the depth of the problem… 😞

Maybe we should rethink the whole "select the whole row to open the item it represents" thing? It feels like we're abusing Tabulator by trying to make it work like a list, rather than a table… 🤔

Or perhaps we could at least decouple the "visually highlight to indicate a row is opened in the related view" from the row selection stuff… 🤔

Reply:
> Personally, I think that clicking a checkbox shouldn't change the status of the related-view at all. Whether checking or unchecking a checkbox, that interaction should neither open nor close the related view, and it also shouldn't change what's shown in the related view, if it's already open and showing something else.

We meant exactly the same. :) Selecting / unselecting checkbox should not affect a related view. The Related view should be changed only if you press on an object (doesn't matter it's checked/selected or not)

Reply:
> Sounds to me like you're contradicting yourself here, so either I misunderstood what I read, or you didn't write what you meant 😄

Well, what I wrote is certainly not clear 😄 sorry for that. But if you look at the picture below (from the demo screen recording), it seems when a row is selected, it also has the activeRow style, no matter if it is active or not. Two rows cannot be active in the same time, right? 

![image](https://user-images.githubusercontent.com/35954987/149332545-1bfeec58-3e5f-468c-bd14-0402c9042a19.png)

So yes 👇 
> You say that a line should be green when (and only when), that item is open in the related view

So what I mean is like what you mean Adrian.

Reply:
Oh! Sorry, no, I'm confused. I thought `tabulator.selectRow` was involved in `activeRow` too, but I was wrong.

Then there shouldn't be too much of a problem I think…? 😅

Reply:
> Oh! Sorry, no, I'm confused. I thought `tabulator.selectRow` was involved in `activeRow` too, but I was wrong.
> 
> Then there shouldn't be too much of a problem keeping them separate in behaviour, I think…? 😅

I removed all use of Tabulator's own row selection because it does not allow us to keep the selection when the user activates a row.
The reason is the behavior added by this else-block in the SelectRow row click handler that keeps activeRow + selection in sync:
https://github.com/olifolkerd/tabulator/blob/4fc283da83dbf6764a88076c8d808d2bcbc37aee/src/js/modules/SelectRow/SelectRow.js#L133-L134

Doing our own makes things much simpler to keep things separate and also avoid any of the unwanted behaviors from the builtin row selection.